### ISSUE TITLE
docker volume create does not respect tenant_id opt

### ISSUE THREAD

**Issue by NolanChan** [quobyte/docker-volume#31]
Hi, we are currently experiencing some issues trying to associate a new volume with a particular `tenant_id`. When we create a volume and pass in a `tenant_id`, it is not respected and grabs the first `tenant_id` created regardless.

Steps:

1. Create tenant on Quobyte dashboard with hypothetical id `my-volume-id`
2. Remove `QUOBYTE_TENANT_ID` from docker-quobyte.env
3. Create volume through CLI

`docker volume create --driver quobyte --name myVolume --opt tenant_id=my-volume-id`

Expected:
A volume is created with tenant_id `my-volume-id`

Actual:
```
Error response from daemon: create myVolume: VolumeDriver.Create: ENTITY_NOT_FOUND/POSIX_ERROR_NONE: Tenant 'NO-DEFAULT-CHANGE-ME' does not exist. /
```

**Comment by casusbelli** [comment]
Hi!
@NolanChan Could you please provide the corresponding part of the docker-quobyte-plugin service log (e.g. for systemd please check 'journalctl -u docker-quobyte-plugin' with the corresponding time stamps).
Also, please make sure you have the latest release (1.3.1) installed, iirc there was a matching bug in versions prior to 1.2 .

**Comment by NolanChan** [comment]
Here's my `journalctl -u docker-quobyte-plugin`

The last 5 lines are the most relevant, as they are the output to when I ran
`docker volume create --driver quobyte nolan2 --opt tenant_id=03bdd274-babb-4dd6-8b73-74386161819e`

It shows that my `tenant_id` is not being passed properly, and is instead defaulting to whatever `tenant_id` is inside `docker-quobyte.env`

```
ubuntu@ip-172-16-0-98:~$ journalctl -u docker-quobyte-plugin
-- Logs begin at Wed 2018-02-28 02:25:00 UTC, end at Wed 2018-02-28 19:16:44 UTC. --
Feb 28 02:29:38 ip-172-16-0-98 systemd[1]: Started Docker Quobyte Plugin.
Feb 28 02:29:38 ip-172-16-0-98 docker-quobyte-plugin[3539]: 2018/02/28 02:29:38
Feb 28 02:29:38 ip-172-16-0-98 docker-quobyte-plugin[3539]: Variables read:
Feb 28 02:29:38 ip-172-16-0-98 docker-quobyte-plugin[3539]: MAX_FS_CHECKS: 5
Feb 28 02:29:38 ip-172-16-0-98 docker-quobyte-plugin[3539]: MAX_WAIT_TIME: 30
Feb 28 02:29:38 ip-172-16-0-98 docker-quobyte-plugin[3539]: SOCKET_GROUP: root
Feb 28 02:29:38 ip-172-16-0-98 docker-quobyte-plugin[3539]: QUOBYTE_API_URL: http://quobyte.private-wedeploy-dev:7860
Feb 28 02:29:38 ip-172-16-0-98 docker-quobyte-plugin[3539]: QUOBYTE_API_USER: wedeploy
Feb 28 02:29:38 ip-172-16-0-98 docker-quobyte-plugin[3539]: QUOBYTE_MOUNT_PATH: /mnt/nfs
Feb 28 02:29:38 ip-172-16-0-98 docker-quobyte-plugin[3539]: QUOBYTE_MOUNT_OPTIONS: -o user_xattr
Feb 28 02:29:38 ip-172-16-0-98 docker-quobyte-plugin[3539]: QUOBYTE_REGISTRY: quobyte.private-wedeploy-dev:7861
Feb 28 02:29:38 ip-172-16-0-98 docker-quobyte-plugin[3539]: QUOBYTE_TENANT_ID:  28434908-4ced-4352-94db-5f7de709690f
Feb 28 02:29:38 ip-172-16-0-98 docker-quobyte-plugin[3539]: QUOBYTE_VOLUME_CONFIG_NAME: BASE
Feb 28 02:29:38 ip-172-16-0-98 docker-quobyte-plugin[3539]: 2018/02/28 02:29:38 Found Mountpoint: /mnt/nfs
Feb 28 02:34:29 ip-172-16-0-98 docker-quobyte-plugin[3539]: 2018/02/28 02:34:29 lstat /mnt/nfs/88531596d68949884b29c68a25d9b9f3ebffbde01c70d9b7e03dc14dd72694f6: no such file or directory
Feb 28 02:47:51 ip-172-16-0-98 docker-quobyte-plugin[3539]: 2018/02/28 02:47:51 Mounting volume 275322171034066907-275322575979664994 on /mnt/nfs/275322171034066907-275322575979664994
Feb 28 02:47:52 ip-172-16-0-98 docker-quobyte-plugin[3539]: 2018/02/28 02:47:52 Mounting volume 275322171034066907-275322575979664994 on /mnt/nfs/275322171034066907-275322575979664994
Feb 28 02:48:00 ip-172-16-0-98 docker-quobyte-plugin[3539]: 2018/02/28 02:48:00 Mounting volume 275322196593772978-275322585028632458 on /mnt/nfs/275322196593772978-275322585028632458
Feb 28 02:48:00 ip-172-16-0-98 docker-quobyte-plugin[3539]: 2018/02/28 02:48:00 Mounting volume 275322196593772978-275322585028632458 on /mnt/nfs/275322196593772978-275322585028632458
Feb 28 02:48:44 ip-172-16-0-98 docker-quobyte-plugin[3539]: 2018/02/28 02:48:44 Mounting volume 275322171034066907-275322576019780949 on /mnt/nfs/275322171034066907-275322576019780949
Feb 28 02:48:44 ip-172-16-0-98 docker-quobyte-plugin[3539]: 2018/02/28 02:48:44 Mounting volume 275322171034066907-275322576019780949 on /mnt/nfs/275322171034066907-275322576019780949
Feb 28 02:49:55 ip-172-16-0-98 docker-quobyte-plugin[3539]: 2018/02/28 02:49:55 Mounting volume 275322188542929727-275322811227048172 on /mnt/nfs/275322188542929727-275322811227048172
Feb 28 02:49:55 ip-172-16-0-98 docker-quobyte-plugin[3539]: 2018/02/28 02:49:55 Mounting volume 275322203628120389-275322671234552185 on /mnt/nfs/275322203628120389-275322671234552185
Feb 28 02:49:55 ip-172-16-0-98 docker-quobyte-plugin[3539]: 2018/02/28 02:49:55 Mounting volume 275322203628120389-275322671234552185 on /mnt/nfs/275322203628120389-275322671234552185
Feb 28 02:49:55 ip-172-16-0-98 docker-quobyte-plugin[3539]: 2018/02/28 02:49:55 Mounting volume 275322188542929727-275322811227048172 on /mnt/nfs/275322188542929727-275322811227048172
Feb 28 02:50:06 ip-172-16-0-98 docker-quobyte-plugin[3539]: 2018/02/28 02:50:06 Mounting volume 275322171034066907-275322772359347935 on /mnt/nfs/275322171034066907-275322772359347935
Feb 28 02:50:06 ip-172-16-0-98 docker-quobyte-plugin[3539]: 2018/02/28 02:50:06 Mounting volume 275322171034066907-275322772359347935 on /mnt/nfs/275322171034066907-275322772359347935
Feb 28 10:17:25 ip-172-16-0-98 docker-quobyte-plugin[3539]: 2018/02/28 10:17:25 lstat /mnt/nfs/754d1723dc251dc75f5ef3552795a522cb2d5def6985da15423986f3d852fd77: no such file or directory
Feb 28 10:17:26 ip-172-16-0-98 docker-quobyte-plugin[3539]: 2018/02/28 10:17:26 Mounting volume 275378339810775069-larademodata on /mnt/nfs/275378339810775069-larademodata
Feb 28 10:17:40 ip-172-16-0-98 docker-quobyte-plugin[3539]: 2018/02/28 10:17:40 Mounting volume 275378339810775069-larademodata on /mnt/nfs/275378339810775069-larademodata
Feb 28 10:35:10 ip-172-16-0-98 docker-quobyte-plugin[3539]: 2018/02/28 10:35:10 lstat /mnt/nfs/7843cc6bcb1b1099cc92914550fdf6cf7ac478f417f0595f6595b88c6285bc33: no such file or directory
Feb 28 10:35:10 ip-172-16-0-98 docker-quobyte-plugin[3539]: 2018/02/28 10:35:10 Mounting volume 275378339810775069-larademodata on /mnt/nfs/275378339810775069-larademodata
Feb 28 10:35:25 ip-172-16-0-98 docker-quobyte-plugin[3539]: 2018/02/28 10:35:25 lstat /mnt/nfs/461fc73d0e61ad40209b342210f4e94b9c6b2ee80c83317a568d82c3096e3653: no such file or directory
Feb 28 17:57:15 ip-172-16-0-98 docker-quobyte-plugin[3539]: 2018/02/28 17:57:15 lstat /mnt/nfs/nolan: no such file or directory
Feb 28 17:57:15 ip-172-16-0-98 docker-quobyte-plugin[3539]: 2018/02/28 17:57:15 Creating volume nolan
Feb 28 17:57:15 ip-172-16-0-98 docker-quobyte-plugin[3539]: Sending JsonRPC message: {"id":"5577006791947779410","jsonrpc":"2.0","method":"createVolume","params":{"name":"nolan","root_user_id":"root","root_group_id":"root","configuration_name":"BASE","tenant_id":"28434908-4ced-4352-94db-5f7de709690f","retry":"INTERACTIVE"}}
Feb 28 17:57:15 ip-172-16-0-98 docker-quobyte-plugin[3539]: 2018/02/28 17:57:15 Validate mounting volume nolan on /mnt/nfs/nolan
Feb 28 17:57:15 ip-172-16-0-98 docker-quobyte-plugin[3539]: 2018/02/28 17:57:15 Unsuccessful Filesystem Check for /mnt/nfs/nolan after 0 tries
Feb 28 17:57:16 ip-172-16-0-98 docker-quobyte-plugin[3539]: 2018/02/28 17:57:16 Unsuccessful Filesystem Check for /mnt/nfs/nolan after 0 tries
Feb 28 17:57:18 ip-172-16-0-98 docker-quobyte-plugin[3539]: 2018/02/28 17:57:18 Unsuccessful Filesystem Check for /mnt/nfs/nolan after 0 tries
Feb 28 17:57:22 ip-172-16-0-98 docker-quobyte-plugin[3539]: 2018/02/28 17:57:22 Unsuccessful Filesystem Check for /mnt/nfs/nolan after 0 tries
Feb 28 18:00:20 ip-172-16-0-98 docker-quobyte-plugin[3539]: 2018/02/28 18:00:20 lstat /mnt/nfs/inTenant2: no such file or directory
Feb 28 18:00:20 ip-172-16-0-98 docker-quobyte-plugin[3539]: 2018/02/28 18:00:20 Creating volume inTenant2
Feb 28 18:00:20 ip-172-16-0-98 docker-quobyte-plugin[3539]: Sending JsonRPC message: {"id":"4037200794235010051","jsonrpc":"2.0","method":"createVolume","params":{"name":"inTenant2","root_user_id":"root","root_group_id":"root","configuration_name":"BASE","tenant_id":"28434908-4ced-4352-94db-5f7de709690f","retry":"INTERACTIVE"}}
Feb 28 18:00:20 ip-172-16-0-98 docker-quobyte-plugin[3539]: 2018/02/28 18:00:20 Validate mounting volume inTenant2 on /mnt/nfs/inTenant2
Feb 28 18:00:20 ip-172-16-0-98 docker-quobyte-plugin[3539]: 2018/02/28 18:00:20 Unsuccessful Filesystem Check for /mnt/nfs/inTenant2 after 0 tries
Feb 28 18:00:21 ip-172-16-0-98 docker-quobyte-plugin[3539]: 2018/02/28 18:00:21 Unsuccessful Filesystem Check for /mnt/nfs/inTenant2 after 0 tries
Feb 28 18:00:23 ip-172-16-0-98 docker-quobyte-plugin[3539]: 2018/02/28 18:00:23 Unsuccessful Filesystem Check for /mnt/nfs/inTenant2 after 0 tries
Feb 28 18:00:27 ip-172-16-0-98 docker-quobyte-plugin[3539]: 2018/02/28 18:00:27 Unsuccessful Filesystem Check for /mnt/nfs/inTenant2 after 0 tries
Feb 28 18:40:20 ip-172-16-0-98 docker-quobyte-plugin[3539]: 2018/02/28 18:40:20 Mounting volume 275441974415604286-images on /mnt/nfs/275441974415604286-images
Feb 28 18:46:55 ip-172-16-0-98 docker-quobyte-plugin[3539]: 2018/02/28 18:46:55 Mounting volume 275441974415604286-images on /mnt/nfs/275441974415604286-images
Feb 28 18:46:55 ip-172-16-0-98 docker-quobyte-plugin[3539]: 2018/02/28 18:46:55 Mounting volume 275441974415604286-images on /mnt/nfs/275441974415604286-images
Feb 28 18:47:27 ip-172-16-0-98 docker-quobyte-plugin[3539]: 2018/02/28 18:47:27 Mounting volume 275441974415604286-images on /mnt/nfs/275441974415604286-images
Feb 28 18:48:05 ip-172-16-0-98 docker-quobyte-plugin[3539]: 2018/02/28 18:48:05 Mounting volume 275441974415604286-images on /mnt/nfs/275441974415604286-images
Feb 28 18:48:05 ip-172-16-0-98 docker-quobyte-plugin[3539]: 2018/02/28 18:48:05 Mounting volume 275441974415604286-images on /mnt/nfs/275441974415604286-images
Feb 28 18:56:33 ip-172-16-0-98 docker-quobyte-plugin[3539]: 2018/02/28 18:56:33 Mounting volume 275441974415604286-images on /mnt/nfs/275441974415604286-images
Feb 28 18:56:33 ip-172-16-0-98 docker-quobyte-plugin[3539]: 2018/02/28 18:56:33 Mounting volume 275441974415604286-images on /mnt/nfs/275441974415604286-images
Feb 28 19:01:32 ip-172-16-0-98 docker-quobyte-plugin[3539]: 2018/02/28 19:01:32 Mounting volume 275441974415604286-images on /mnt/nfs/275441974415604286-images
Feb 28 19:14:58 ip-172-16-0-98 docker-quobyte-plugin[3539]: 2018/02/28 19:14:58 lstat /mnt/nfs/nolan2: no such file or directory
Feb 28 19:14:58 ip-172-16-0-98 docker-quobyte-plugin[3539]: 2018/02/28 19:14:58 Creating volume nolan2
Feb 28 19:14:58 ip-172-16-0-98 docker-quobyte-plugin[3539]: Sending JsonRPC message: {"id":"605394647632969758","jsonrpc":"2.0","method":"createVolume","params":{"name":"nolan2","root_user_id":"root","root_group_id":"root","configuration_name":"BASE","tenant_id":"28434908-4ced-4352-94db-5f7de709690f","retry":"INTERACTIVE"}}
Feb 28 19:14:59 ip-172-16-0-98 docker-quobyte-plugin[3539]: 2018/02/28 19:14:59 Validate mounting volume nolan2 on /mnt/nfs/nolan2
Feb 28 19:14:59 ip-172-16-0-98 docker-quobyte-plugin[3539]: 2018/02/28 19:14:59 Unsuccessful Filesystem Check for /mnt/nfs/nolan2 after 0 tries
Feb 28 19:15:00 ip-172-16-0-98 docker-quobyte-plugin[3539]: 2018/02/28 19:15:00 Unsuccessful Filesystem Check for /mnt/nfs/nolan2 after 0 tries
lines 20-65/65 (END)
```

**Comment by NolanChan** [comment]
We are using version `1.3.1`

**Comment by casusbelli** [comment]
Fix released in 1.3.2
